vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -289,8 +316,16 @@ def __init__(
 
         self.tp_size = (tp_size if tp_size is not None else
                         get_tensor_model_parallel_world_size())
+        # Chendi: Temporarily add ep_support using parameter
+        # It should be gone once version updated to upstream
+        self.ep_size = self.tp_size \
+            if envs.VLLM_ENABLE_EXPERT_PARALLEL else 1
+        self.tp_size = self.tp_size // self.ep_size
+        assert num_experts % self.ep_size == 0
+        self.ep_rank = get_tensor_model_parallel_rank() // self.tp_size
+
         self.top_k = top_k
-        self.num_experts = num_experts
+        self.num_experts = num_experts // self.ep_size
```

Comment:
All above is temporary fix because habana-main is too lagging behind and missing important commits from upstream for Expert parallelism